File: app/models/research.py
from pydantic import BaseModel

class ResearchRequest(BaseModel):
    query: str
    

class VerificationSummary(BaseModel):
    SUPPORTED: int
    DISPUTED: int
    UNCERTAIN: int
    avg_confidence: float
    total_claims: int   

class ResearchResponse(BaseModel):
    answer: str
    research_results: list[dict]
    extracted_claims: list[dict]
    verifications: list[dict]
    contradictions: list[dict]
    topic_clusters: dict
    verification_summary: VerificationSummary
    claims_stored: int
    similar_past_claims: list[dict]
    all_sources: list[str]
    session_id: str
 
    model_config = {"extra": "ignore"}

File: frontend/types/research.ts
export interface ResearchResult {
  question: string;
  summary: string;
  sources: string[];
  confidence: number;
}

export interface Verification {
  claim_id: string;
  claim: string;
  status: "SUPPORTED" | "DISPUTED" | "UNCERTAIN";
  supporting_evidence: string[];
  contradicting_evidence: string[];
  confidence: number;
  reason: string;
  ml_stance: "SUPPORTS" | "OPPOSES" | "NEUTRAL";
  ml_stance_confidence: number;
  ml_similarity: number;
  ml_confidence: number;
}

export interface Contradiction {
  claim_a: string;
  claim_b: string;
  topic: string;
  relation: "CONTRADICTS" | "SUPPORTS" | "UNRELATED";
  explanation: string;
}

export interface VerificationSummary {
  SUPPORTED: number;
  DISPUTED: number;
  UNCERTAIN: number;
  avg_confidence: number;
  total_claims: number;
  ml_stance_counts: {
    SUPPORTS: number;
    OPPOSES: number;
    NEUTRAL: number;
  };
}

export interface ResearchResponse {
  answer: string;
  research_results: ResearchResult[];
  extracted_claims: any[];
  verifications: Verification[];
  contradictions: Contradiction[];
  topic_clusters: Record<string, string[]>;
  ml_candidate_pairs: any[];
  verification_summary: VerificationSummary;
  claims_stored: number;
  similar_past_claims: any[];
  all_sources: string[];
  session_id: string;
}

export interface TraceStep {
  label: string;
  detail: string;
  done: boolean;
}